pkg/resource/vsc/scan.go

```diff
@@ -27,9 +32,15 @@ func (s *Scan) UM(duration time.Duration) (resource.UMMeasurement, error) {
 }
 
 func (s *Scan) EDP() (resource.EDPMeasurement, error) {
+	errs := []error{}
 	edp := resource.EDPMeasurement{}
 
 	for _, vsc := range s.vscs.Items {
+		if vsc.Status == nil {
+			errs = append(errs, fmt.Errorf("%w: %s", ErrStatusNotSet, vsc.Name))
+			continue
+		}
+
 		if vsc.Status.ReadyToUse != nil && *vsc.Status.ReadyToUse {
 			currVSC := getSizeInGB(*vsc.Status.RestoreSize)
```

Comment:
Can we also check that the `vsc.Status.RestoreSize` is not nil before dereferencing?
Since it is also an optional field and in the documentation it is mentioned `If not specified, it indicates that the size is unknown.`

Reply:
Will throw an error in this situation. 
We will not throw an error if ready to use is not set, as this seems rather "normal" for us.